### DISCUSSION TITLE
ESP32-C3 support

### DISCUSSION THREAD

**Discussion by MicroMontis** [ExpressLRS/ExpressLRS#2507] [Ideas]
It would be great if ESP32-C3 would be supported. I'm thinking about the super easy DIY RX (or also TX) using just one bare ESP32-C3 SuperMini dev board + E28-2G4M12SX. I think you would only need these two ready made modules on some PCB and it would make the easiest DIY ELRS RX in the world!

![image](https://github.com/ExpressLRS/ExpressLRS/assets/15938741/8f62c574-11f5-4e72-8547-8f39aeaad6ab)

Please consider supporting the ESP32-C3.

**Comment by Linjieqiang** [comment]
Any plan?

**Comment by pkendall64** [comment]
We have a testing branch underway now. It's looking promising and will make a nice replacement for the 8285 based receivers.

**Reply by Linjieqiang** [reply]
Can it support transmitters?

**Reply by pkendall64** [reply]
Only very basic ones because theres a very limited number of GPIO available. I have not configured the code for that use case.

**Reply by winiak** [reply]
I'm looking forward for it as well. Perfect for small DIY project!

**Reply by n0kkster** [reply]
hello, is it possible to download this branch now and test by myself?

**Reply by pkendall64** [reply]
Of course, just clone the repository and checkout the PR using github-desktop or similar.

**Comment by 0990** [comment]
I have make one(esp32c3supermini+A28-S2G4A12s4a) and boot with some error:1,cpu cannot work at 240MHz,esp32c3  only work at 160MHz,2,'generic esp32 c3 rx' default config,led is rgb,but I want to use supermini led,which is gpio8.I fix above issues and it boot success, LED Indication:Double blink then pause,but it cann't bind.maybe my A28-S2G4A12s4a is broken,I will make new one and feedback soon.

**Reply by qingwa2009** [reply]
你那个c3supermini 发射机能绑定了没有？我折腾了好久都绑定不了，按了进入绑定模式，过个半秒就自动退出绑定模式了

**Reply by 0990** [reply]
我的可以的啊，注意supermini版本，有的led是彩色，有的不是

**Reply by 0990** [reply]
[Supermini C3 2400 RGB PWM.json](https://github.com/user-attachments/files/23568809/Supermini.C3.2400.RGB.PWM.json)
[Supermini C3 2400.json](https://github.com/user-attachments/files/23568810/Supermini.C3.2400.json)
我上传了两个配置，一个是RGB的灯，输出是PWM;一个是单色灯，输出是uart,你看着改改看，注意在targets.json里要添加个配置，这样手动编译upload时才能找到

**Comment by 0990** [comment]
It worked when I checkout 3.5.4 version, the pcb is here: https://oshwhub.com/jjjkkkkkk/elrs-jie-shou-ji-esp32_c3-fang-an
